src/tree/v2/workspace/WorkspaceResourceTreeDataProvider.ts

```diff
@@ -16,14 +16,13 @@ export class WorkspaceResourceTreeDataProvider extends ResourceTreeDataProviderB
     constructor(
         private readonly branchDataProviderManager: WorkspaceResourceBranchDataProviderManager,
         onRefresh: vscode.Event<void | ResourceGroupsItem | ResourceGroupsItem[] | null | undefined>,
-        private readonly resourceProviderManager: WorkspaceResourceProviderManager) {
+        private readonly resourceProviderManager: WorkspaceResourceProviderManager,
+        itemCache: BranchDataItemCache) {
```

Comment:
Could you explain why the workspace tree view needs the `itemCache` now? I don't really have an opposing opinions on it, just wondering if it's only for reveal and if it's something that we'll remove in the future.

Reply:
The workspace tree already had an item cache. I've just moved where it's instantiated so that I can use it for reveal as well.

Reply:
> EDIT: Never mind, I re-read your PR description and it makes sense. But maybe make a note to remove it from the `TreeDataProvider` as well?

In the tree data provider the item cache is used for more than just reveal. It's also used for refreshing, since we need to map branch data items into resource groups items to pass to `onDidChangeTreeData`.